compiler/next/lib/frontend/Parser/chapel.ypp

```diff
@@ -2511,18 +2511,78 @@ new_maybe_decorated:
 ;
 
 new_expr:
-  /* handles the typical new cases, e.g. new C(); new owned C() */
+  /* Handles the typical new cases, e.g. new (); new owned C() */
   new_maybe_decorated expr %prec TNEW
-    { TODOEXPR(@$); }
-  /* handles e.g. new (typefn())(initargs) */
+  {
+    auto management = New::Management::None;
+    switch ($1) {
+      case NewTag_BORROWED: management = New::Management::Borrowed; break;
+      case NewTag_OWNED: management = New::Management::Owned; break;
+      case NewTag_SHARED: management = New::Management::Shared; break;
+      case NewTag_UNMANAGED: management = New::Management::Unmanaged; break;
+      case NewTag_BLANK: management = New::Management::None; break;
+    }
+
+    Call* fnCall = $2->toFnCall();
+
+    // If the type expression is a call that uses parens and has a base
+    // expression, we can modify it in place. What we will do is wrap the
+    // base expression of the call in a 'New'.
+    if (fnCall && fnCall->calledExpression()) {
```

Comment:
this seems complex enough to justify moving it to a method on ParserContext. It's better to have less code in the chapel.ypp so we don't have to rebuild it if we modify that code.